### PR TITLE
feat: Launch JupyterLab" button should appear on tasks page [WEB-811]

### PR DESCRIPTION
**Pull request by julian-determined-ai**
## Description

feat: Launch JupyterLab" button should appear on tasks page [WEB-811]

Updates:

1. Added `canCreateNSC` and `canCreateWorkspaceNSC` methods.
2.  Added required `Workspace` dropdown to the JupyterLab "Full Configuration" form.
3.  Added required `Workspace` dropdown to the JupyterLab "Simple Configuration" form.
4. Added ability to specify a workspace in `useModalJupyterLab` that will disable and preselect a value for  the dropdown in (2) and (3).
5. Removed the feature flag check that was previously hiding the `Launch JupyterLab` button on the `Tasks` page.

Note: The regular `Launch JupyterLab button` on the navigation sidebar has remained unchanged, in terms of being disabled/enabled. If we also need to change that buttons behavior then that can be done.

## Test Plan

1. Navigate to the `Tasks` page. 
2. Click the "Launch JupyterLab" button. 
3. Click `Launch` without specifying a workspace. Ensure you see the error message `Workspace is required`.
4. Click the Workspace dropdown and choose a workspace. 
5. Click "Launch" and ensure you are taken to the interactive tasks page. 
6. Navigate to the `Tasks` page. 
7.  Click the "Launch JupyterLab" button. 
8. Click the "Show Full Config" button. 
9. Click `Launch` without specifying a workspace. Ensure you see the error message `Workspace is required`.
10. Click the Workspace dropdown and choose a workspace. 
11. Click "Launch" and ensure you are taken to the interactive tasks page. 
12. In both the "Simple" and "Full" configuration forms ensure that changing values and submitting the form works as expected.
13. Navigate to a workspace page then the "Tasks" tab for the workspace. Click the "Launch JupyterLab" button. Ensure that both in the "Full" and "Simple" configuration the Workspace drowpdown is disabled and preselected to the current workspace. Click "Launch" and ensure the task shows up in the workspaces' "Tasks" list.
15. (Easiest tested with RBAC enabled) Navigate to the Tasks page or to a workspace where you do not have NSC permissions. Ensure the `Launch JupyterLab` button is disabled and a tooltip is shown.

4CC3AA16-3CE3-4A2F-B811-733EAB10F038

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

WEB-811 

[WEB-811]: https://determinedai.atlassian.net/browse/WEB-811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by hamidzr** [comment]
from the description I couldn't make out if we want this part included or not but: the Jupiter launch button on the tasks page doesn't seem to specify the workspace id so everything ends up launching in the default workspace 
image

**Comment by julian-determined-ai** [comment]
> thanks for working on this. from the description I couldn't make out if we want this part included or not but: the Jupiter launch button on the tasks page doesn't seem to specify the workspace id so everything ends up launching in the default workspace image

For now I believe this is oaky. This case should be handled in https://determinedai.atlassian.net/browse/WEB-810, where a user will not even be shown the Task Page unless they have relevant permissions to create a task in at least one workspace. At that point, the tasks page will only show the correct workspace id. 

Additionally, the reason why no workspace is needed in this PR is because the update to the button in the task page will not happen until https://github.com/determined-ai/determined/pull/5782 is completed.

File: webui/react/src/components/NavigationSideBar.tsx [hunk 1]
```diff
@@ -258,13 +258,20 @@ const NavigationSideBar: React.FC = () => {
             <section className={css.launch}>
               <div className={css.launchBlock}>
                 <div className={css.launchButton}>
-                  <Button block type="ghost" onClick={() => openJupyterLabModal()}>
+                  <Button
+                    block
+                    disabled={!canCreateNSC}
+                    type="ghost"
```
Comment (mapmeld) [inline webui/react/src/components/NavigationSideBar.tsx:264]: Console says `Warning: [antd: Button] `link` or `text` button can't be a `ghost` button.`

File: webui/react/src/components/TaskList.tsx [hunk 2]
```diff
@@ -552,7 +558,13 @@ const TaskList: React.FC<Props> = ({ workspaceId }: Props) => {
   );
 
   const JupyterLabButton = () => {
-    return <Button onClick={() => openJupyterLabModal()}>Launch JupyterLab</Button>;
+    return (
+      <Button
+        disabled={workspace ? !canCreateWorkspaceNSC(workspace) : !canCreateNSC}
```
Comment (mapmeld) [inline webui/react/src/components/TaskList.tsx:563]: you're probably on this already but I had to change this to `{ workspace }` to run

**Comment by julian-determined-ai** [comment]
@mapmeld I reassigned this to myself now that the other PR has merged in and I can make updates id hold off on commenting for a bit. But feel free to, but things will likely be changed.

**Review by mapmeld — APPROVED** [review]
Worked during earlier testing, LGTM

**Pull request merged**